ctapipe/core/tests/test_container.py

```diff
@@ -236,26 +236,32 @@ def test_field_validation():
     field_f.validate(np.ones((2, 2), dtype=np.float64))
 
     #   test dtype
+    field_f.validate(np.ones((2, 2), dtype=np.float32))
+    field_f.validate(np.ones((2, 2), dtype=np.int32))
     with pytest.raises(FieldValidationError):
```

Comment:
`pytest.raises` has the `match` keyword, this allows us to match on `ndarray`, `dimensionality` and `dtype`, depending on why these tests should fail.  I think this increases the ease of understanding these tests.